File: rb/features.rb
# Aareguru SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AareguruFeatures
  def self.make_feature(name)
    case name
    when "base"
      AareguruBaseFeature.new
    when "ratelimit"
      AareguruRatelimitFeature.new
    when "retry"
      AareguruRetryFeature.new
    when "test"
      AareguruTestFeature.new
    when "timeout"
      AareguruTimeoutFeature.new
    else
      AareguruBaseFeature.new
    end
  end
end

File: py/features.py
# Aareguru SDK feature factory

from feature.base_feature import AareguruBaseFeature
from feature.test_feature import AareguruTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AareguruBaseFeature(),
        "test": lambda: AareguruTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
